src/workflows-app/components/SubmitWorkflowModal.test.ts

```diff
@@ -0,0 +1,385 @@
+import { getTerraUser, workflowsAppStore } from 'src/libs/state';
+import { asMockedFn, renderWithAppContexts as render } from 'src/testing/test-utils';
+import { SubmitWorkflowModal } from 'src/workflows-app/components/SubmitWorkflowModal';
+import { methodDataWithVersions } from 'src/workflows-app/utils/mock-data';
+import {
+  mockAzureApps,
+  mockAzureWorkspace,
+  mockCollaborativeAzureApps,
+  mockCromwellRunner,
+  runSetInputDef,
+  runSetOutputDefFilled,
+} from 'src/workflows-app/utils/mock-responses';
+
+jest.mock('src/libs/ajax');
+
+type AjaxContract = ReturnType<typeof Ajax>;
+
+jest.mock('src/libs/notifications.js');
+
+jest.mock('src/libs/config', () => ({
+  ...jest.requireActual('src/libs/config'),
+  getConfig: jest.fn().mockReturnValue({}),
+}));
+
+jest.mock('src/libs/nav', () => ({
+  getCurrentUrl: jest.fn().mockReturnValue(new URL('https://app.terra.bio')),
+  goToPath: jest.fn(),
+}));
+
+jest.mock('src/libs/state', () => ({
+  ...jest.requireActual('src/libs/state'),
+  getTerraUser: jest.fn().mockReturnValue({ id: 'foo' }),
+}));
+
+jest.mock('src/libs/utils', () => ({
+  ...jest.requireActual('src/libs/utils'),
+  poll: jest.fn(async (fn) => {
+    // remove delay
+    while (true) {
+      const r = await fn();
+      if (!r.shouldContinue) {
+        return r.result;
+      }
+    }
+  }),
+}));
+
+jest.mock('src/components/Modal', () => {
+  const mockModal = jest.requireActual('src/components/Modal.mock');
+  return mockModal.mockModalModule();
+});
+
+jest.mock('src/libs/ajax/metrics/useMetrics', () => ({
+  ...jest.requireActual('src/libs/ajax/metrics/useMetrics'),
+  useMetricsEvent: jest.fn(() => ({ captureEvent: jest.fn() })),
+}));
+
+jest.mock('src/libs/feature-previews', () => ({
+  ...jest.requireActual('src/libs/feature-previews'),
+  isFeaturePreviewEnabled: jest.fn(),
+}));
+
+const submitModalProps = {
+  method: methodDataWithVersions.methods[0],
+  methodVersion: methodDataWithVersions.methods[0].method_versions[0],
+  recordType: 'FOO',
+  selectedRecords: { FOO1: 'bar' },
+  inputDefinition: runSetInputDef,
+  outputDefinition: runSetOutputDefFilled,
+  callCachingEnabled: true,
+  onDismiss: jest.fn(),
+  name: 'test-azure-ws-name',
+  namespace: 'test-azure-ws-namespace',
+  workspace: mockAzureWorkspace,
+};
+
+const submitModalPropsReader = {
+  ...submitModalProps,
+  workspace: {
+    ...mockAzureWorkspace,
+    canCompute: false,
+  },
+};
+
+const postRunSetPayload = expect.objectContaining({
+  method_version_id: methodDataWithVersions.methods[0].method_versions[0].method_version_id,
+  workflow_input_definitions: runSetInputDef,
+  workflow_output_definitions: runSetOutputDefFilled,
+  wds_records: {
+    record_type: 'FOO',
+    record_ids: ['FOO1'],
+  },
+  call_caching_enabled: true,
+});
+
+describe('SubmitWorkflowModal submitting to cromwell', () => {
+  beforeEach(() => {
+    workflowsAppStore.reset();
+    asMockedFn(getConfig).mockReturnValue({
```

Comment:
This is ok, but seems to be testing the "dev mode" version of the UI where we fetch URLs from config rather than from Leo. I guess that's ok, but if it's an easy-swap-out, mocking the response from Leo instead might be even nicer.

Reply:
Will look into this